python/lsst/daf/butler/cli/log.py

```diff
@@ -0,0 +1,160 @@
+log4j.appender.A1=ConsoleAppender
+log4j.appender.A1.Target=System.err
+log4j.appender.A1.layout=PatternLayout
+log4j.appender.A1.layout.ConversionPattern={}
+"""
+
+
+class Log:
+
+    _initialized = False
+
+    longLogFmt = "%-5p %d{yyyy-MM-ddTHH:mm:ss.SSSZ} %c (%X{LABEL})(%F:%L)- %m%n"
+    normalLogFmt = "%c %p: %m%n"
+
+    @staticmethod
+    def getPyLogLevel(level):
+        """Get the numeric value for the given log level name.
+
+        Parameters
+        ----------
+        level : `str`
+            One of the python `logging` log level names.
+
+        Returns
+        -------
+        numericValue : `int`
+            The python `logging` numeric value for the log level.
+        """
+        return getattr(logging, level, None)
+
+    @staticmethod
+    def getLsstLogLevel(level):
+        """Get the numeric value for the given log level name.
+
+        If `lsst.log` is not setup this function will return `None` regardless
+        of input. `daf_butler` does not directly depend on `lsst.log` and so it
+        will not be setup when `daf_butler` is setup. Packages that depend on
+        `daf_butler` and use `lsst.log` may setup `lsst.log`.
+
+        Will adapt the python name to an `lsst.log` name:
+        - CRITICAL to FATAL
+        - WARNING to WARN
+
+        Parameters
+        ----------
+        level : `str`
+            One of the python `logging` log level names.
+
+        Returns
+        -------
+        numericValue : `int` or `None`
+            The `lsst.log` numeric value.
+        """
+        if lsstLog is None:
+            return None
+        if level == "CRITICAL":
+            level = "FATAL"
+        elif level == "WARNING":
+            level = "WARN"
+        return getattr(lsstLog.Log, level, None)
+
+    @classmethod
+    def initLog(cls, longlog):
+        """Initialize logging. This should only be called once per program
+        execution. After the first call this will log a warning and return.
+
+        Parameters
+        ----------
+        longlog : `bool`
+            If True, make log messages appear in long format, by default False.
+        """
+        if cls._initialized:
+            # Unit tests that execute more than one command do end up
+            # calling this fucntion multiple times in one program execution,
+            # so do warn but don't log an error or fail, just make the
+            # re-initalization a no-op.
+            log = logging.getLogger(__name__.partition(".")[2])
+            log.warning("Log is already initialized, returning without re-initializing.")
+            return
+        cls._initialized = True
+
+        lgr = logging.getLogger()
+        if lsstLog is not None:
+            # global logging config
+
+            lsstLog.configure_prop(_LOG_PROP.format(cls.longLogFmt if longlog else cls.normalLogFmt))
+
+            # Forward all Python logging to lsstLog
+            lgr.addHandler(lsstLog.LogHandler())
+        lgr.setLevel(logging.INFO)
```

Comment:
If `lsstLog` is `None` then I think we probably need to call `logging.basicConfig(level=INFO)` instead of just `lgr.setLevel(logging.INFO)`.